src/backend/env/user.rs

```diff
@@ -611,6 +611,21 @@ impl User {
             Ok(())
         })
     }
+
+    pub fn validate_send_credits(&self, state: &State) -> Result<(), String> {
+        if let Some(invited_by) = self.invited_by {
+            let invite = state.invite_codes.values().find(|invite| {
+                invited_by == invite.user_id && invite.joined_user_ids.contains(&self.id)
+            });
+            if let Some(invite) = invite {
+                if self.credits_burned() < invite.credits_per_user && self.credits() < 1000 {
```

Comment:
The first condition is an excellent idea!!!

But why do we need the check `< 1000`?

Also, what's the life cycle of an invite? I think the old ones get deleted once used.

Reply:
Removed it. It was supposed to allow users that purchased credits, but attacker can simply send 1000 to all accounts and send 1050 back. 

Invite doesn't get removed anymore, it can be topped up when it reaches 0 credits. Use case will be:

- print qr code and paste on university/town-hall wall
- top it up based on how promotion is going